src/coreComponents/physicsSolvers/contact/SolidMechanicsAugmentedLagrangianContact.cpp

```diff
@@ -640,71 +719,128 @@ bool SolidMechanicsAugmentedLagrangianContact::updateConfiguration( DomainPartit
       arrayView1d< real64 const > const & normalTractionTolerance =
         subRegion.getReference< array1d< real64 > >( viewKeyStruct::normalTractionToleranceString() );
 
+      arrayView1d< real64 const > const area = subRegion.getElementArea().toViewConst();
+
       std::ptrdiff_t const sizes[ 2 ] = {subRegion.size(), 3};
       traction_new.resize( 2, sizes );
       arrayView2d< real64 > const traction_new_v = traction_new.toView();
 
       // TODO: Create a Kernel to update Traction
       forAll< parallelHostPolicy >( subRegion.size(),
-                                    [ traction_new_v, traction, penalty, dispJump, deltaDispJump ] ( localIndex const kfe )
+                                    [ traction_new_v, traction, penalty, 
+                                      dispJump, deltaDispJump, fractureState ] ( localIndex const kfe )
       {
         real64 eps_N = penalty[kfe][0];
         real64 eps_T = penalty[kfe][1];
         traction_new_v[kfe][0] = traction[kfe][0] + eps_N * dispJump[kfe][0];
-        traction_new_v[kfe][1] = traction[kfe][1] + eps_T * deltaDispJump[kfe][1];
-        traction_new_v[kfe][2] = traction[kfe][2] + eps_T * deltaDispJump[kfe][2];
+        traction_new_v[kfe][1] = traction[kfe][1];
+        traction_new_v[kfe][2] = traction[kfe][2];
+        if (fractureState[kfe] == contact::FractureState::Stick)
+        {
+          traction_new_v[kfe][1] +=  eps_T * deltaDispJump[kfe][1];
+          traction_new_v[kfe][2] +=  eps_T * deltaDispJump[kfe][2];
+        }
       } );
 
-      forAll< parallelHostPolicy >( subRegion.size(),
-                                    [ &condConv, ghostRank, traction_new_v, normalTractionTolerance, normalDisplacementTolerance, slidingTolerance, deltaDispJump, dispJump ] ( localIndex const kfe )
+      
+      real64 const slidingCheckTolerance = m_slidingCheckTolerance;
+      constitutiveUpdatePassThru( contact, [&] ( auto & castedContact )
       {
-        if( ghostRank[kfe] < 0 )
+        using ContactType = TYPEOFREF( castedContact );
+        typename ContactType::KernelWrapper contactWrapper = castedContact.createKernelWrapper();
+
+        bool printflag = true;
+
+        forAll< parallelHostPolicy >( subRegion.size(),
```

Comment:
why not `parallelDevicePolicy`? You won't be able to capture anything by reference. And you won't be able to capture `this` in a device lambda coz everything must be capture by value and it will basically copy a host address and try to use it on device...

Reply:
This was only for debugging purposes. I needed to print inside a RAJA loop, and this was the fastest method I thought of. Without passing `this` I can't use `GEOS_FMT`. As soon as everything works I will remove `this` and I will switch to `parallelDevicePolicy`.